crates/autopilot/src/run.rs

```diff
@@ -508,6 +520,7 @@ pub async fn run(args: Arguments) {
             .iter()
             .cloned()
             .collect::<HashSet<_>>(),
+        cow_amm_indexer,
```

Comment:
We could have a `struct` which wraps the indexer to add another layer of abstraction.

Reply:
For the wrapper I am thinking of either:
- share the `index` object to get the updates
- make a mscp and have a listener of the `index`
- other?

I am open to any suggestion.

Reply:
Follow up PR addressing this issue here: https://github.com/cowprotocol/services/pull/2776
It also creates the placeholder for the other event listeners for CoW AMM events.